[explore] viz type selector as modal

mistercrunch opened the pull request:
![img](http://g.recordit.co/x0kIKQc2bO.gif)

mistercrunch committed:
[explore] viz type selector as modal

ascott commented on superset/assets/javascripts/explorev2/components/controls/VizTypeControl.jsx:
```diff
@@ -0,0 +1,95 @@
+    return (
+      <div>
+        <Label>{visTypes[this.props.value].label}</Label>
+        <Button onClick={this.toggleModal} bsSize="sm" className="l-m-5">
```
could we stick to a similar format for editing controls, and add an edit link rather than the button with viz types?

<img width="890" alt="screenshot 2017-05-23 11 52 28" src="https://cloud.githubusercontent.com/assets/130878/26371325/44cd76ea-3faf-11e7-8990-0696cfa4ef2e.png">

ascott commented on superset/assets/javascripts/explorev2/components/controls/VizTypeControl.jsx:
```diff
@@ -0,0 +1,95 @@
+                      className={`viztype-selector ${this.props.value === vt ? 'selected' : ''}`}
+                      src={`/static/assets/images/viz_thumbnails/${vt}.png`}
+                    />
+                    <div className="viztype-label">
```
- could we add the label under the image, rather than over it? we should allow long vis type names to wrap on to two lines. 
- could also get rid of the box shadow on the boxes, since we stopped using them in other places as well.
- can we make the vis type boxes span the modal 100%, to match the width of the search box?
<img width="896" alt="screenshot 2017-05-23 12 05 31" src="https://cloud.githubusercontent.com/assets/130878/26371683/54190492-3fb0-11e7-8ce8-d300ad24912f.png">

ascott commented on superset/assets/package.json:
```diff
@@ -98,9 +99,10 @@
   "devDependencies": {
     "babel-cli": "^6.14.0",
     "babel-core": "^6.10.4",
+    "babel-istanbul": "^0.12.2",
```
what is this change for?

ascott commented on superset/assets/javascripts/explorev2/main.css:
```diff
@@ -34,3 +34,39 @@
 .background-transparent {
   background-color: transparent !important;
 }
+.viztype-selector-container {
```
we could use react-bootstrap's grid system for layout and probably get rid of some of this css
https://react-bootstrap.github.io/components.html#grid

mistercrunch committed:
Addressing comments

mistercrunch commented:
@ascott addressed your comments:
<img width="744" alt="screen shot 2017-05-25 at 12 12 50 pm" src="https://cloud.githubusercontent.com/assets/487433/26466399/8920ff98-4143-11e7-9370-01e5029d75c5.png">
<img width="426" alt="screen shot 2017-05-25 at 12 12 43 pm" src="https://cloud.githubusercontent.com/assets/487433/26466400/8921d9e0-4143-11e7-909a-d4728ad4a518.png">

ascott commented on superset/assets/javascripts/explore/components/controls/VizTypeControl.jsx:
```diff
@@ -0,0 +1,108 @@
+import ControlHeader from '../ControlHeader';
+
+const propTypes = {
+  description: PropTypes.string,
```
if description and label are not required, we should add defaults for them.

mistercrunch replied:
If they aren't defined we don't print them, that's the case for `FilterControl`.

ascott commented on superset/assets/javascripts/explore/main.css:
```diff
@@ -34,3 +34,32 @@
   background-color: transparent !important;
 }
+
+a {
```
this is pretty general, seems like it should go in the theme css file. 
https://github.com/airbnb/superset/blob/master/superset/assets/stylesheets/less/cosmo/bootswatch.less

mistercrunch replied:
Yes the previous solution to get a cursor was to add an `href='#'` prop and this css is better IMO, will move to the theme

ascott commented:
thanks for making those changes. a couple small comments, otherwise LGTM!

mistercrunch committed:
Adressing comments
